spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestStoragePartitionedJoinsInRowLevelOperations.java

```diff
@@ -134,11 +134,16 @@ private void checkDelete(RowLevelOperationMode mode) {
           SparkPlan plan =
               executeAndKeepPlan(
                   "DELETE FROM %s t WHERE "
-                      + "EXISTS (SELECT 1 FROM %s s WHERE t.id = s.id AND t.dep = s.dep) AND "
-                      + "dep = 'hr'",
```

Comment:
We used to have a filter `dep = 'hr`. The optimizer got smarter and infers an extra filter for the other table and removes `t.dep = s.dep` from the join condition. The last step breaks SPJ as `dep` is a partition column. There is no partition columns in the join condition.